drivers/soundwire/stream.c

```diff
@@ -1403,8 +1403,15 @@ static int _sdw_prepare_stream(struct sdw_stream_runtime *stream,
 	struct sdw_bus *bus;
 	struct sdw_master_prop *prop;
 	struct sdw_bus_params params;
+	int state = stream->state;
 	int ret;
 
+	/* first mark the state as PREPARED so that bus->compute_params() can distinguish
+	 * the stream that is CONFIGURED and PREPARED and skip the stream that is CONFIGURED
+	 * but not in the preparing stage.
+	 */
+	stream->state = SDW_STREAM_PREPARED;
```

Comment:
I think in other patches we also moved the set to PREPARED state prior that we actually prepare things.

I got the reason and at the same time it sounds wrong to set something as PREPARED when it is not...

Reply:
I agree that the stream->state may not represent the actual state perfectly. But, I can't find a better way to skip the m_rt that should not be counted yet. And stream->state is used by the stream helpers only and the helpers will be called one by one. So, I think it is safe to change the stream->state earlier in the same stream helper.